File: data/EXP01/golang_no_system_var/2178/2178.output_3.go
func maximumEvenSplit(finalSum int64) []int64 {
    result := []int64{}

    evenNumber := finalSum / 2 * 2
    result = append(result, evenNumber)

    for finalSum > 0 && finalSum%2 == 0 {
        finalSum -= evenNumber
        evenNumber = finalSum / 2 * 2
        result = append(result, evenNumber)
    }

    if finalSum == 0 {
        return result
    }

    return []int64{}
}

File: data/EXP01/python/2178/2178.output_3.py
# @lc app=leetcode id=2178 lang=python3
class Solution:
    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        result = []
        remainingSum = finalSum

        for i in range(2, finalSum + 1, 2):
            if remainingSum >= i:
                result.append(i)
                remainingSum -= i
            else:
                break

        if remainingSum == 0:
            return result
        else:
            return []